### PR TITLE
feat(components/tabs): add descriptive aria label to tab buttons

### PR DESCRIPTION
**Pull request by Blackbaud-ErikaMcVey**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/586/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/586/storybooks/ag-grid-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/586/storybooks/indicators-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/586/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/586/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/586/storybooks/pages-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/586/storybooks/popovers-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/586/storybooks/tabs-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/586/storybooks/tiles-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/586/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/586/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/586/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/586/playground/)

**Review by bradenbiz — DISMISSED** [review]

File: libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts [hunk 1]
```diff
@@ -34,7 +36,13 @@ const DEFAULT_ELEMENT_ROLE = 'tab';
 })
 export class SkyTabButtonComponent implements AfterViewInit, OnDestroy {
   @Input()
-  public active: boolean;
+  public get active(): boolean {
+    return this.#_isActive;
+  }
+  public set active(value: boolean) {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts:43]: `boolean | undefined`

File: libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts [hunk 2]
```diff
@@ -55,11 +63,24 @@ export class SkyTabButtonComponent implements AfterViewInit, OnDestroy {
   public closeable: boolean;
 
   @Input()
-  public disabled: boolean;
+  public get disabled(): boolean {
+    return this.#_isDisabled;
+  }
+
+  public set disabled(value: boolean) {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts]: `boolean | undefined`

File: libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts [hunk 3]
```diff
@@ -55,11 +63,24 @@ export class SkyTabButtonComponent implements AfterViewInit, OnDestroy {
...
+  }
 
   @Input()
   public tabIndex: SkyTabIndex;
 
+  @Input()
+  public tabNumber: number;
+
+  @Input()
+  public totalTabsCount: number;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts]: We can just do this for the new inputs (for now), need `| undefined`.

File: libs/components/tabs/src/lib/modules/tabs/tabset.component.spec.ts [hunk 4]
```diff
@@ -1621,6 +1622,51 @@ describe('Tabset component', () => {
...
+        const tabBtns = wizardFixture.debugElement.queryAll(
+          By.css('.sky-btn-tab')
+        );
+        const tabBtn1 = tabBtns[0]?.nativeElement;
+        const tabBtn2 = tabBtns[1]?.nativeElement;
+        const tabBtn3 = tabBtns[2]?.nativeElement;
+
+        expect(tabBtn1?.ariaLabel).toEqual('Step 1 of 3, completed: Step 1');
+        expect(tabBtn2?.ariaLabel).toEqual('Step 2 of 3, current: Step 2');
+        expect(tabBtn3?.ariaLabel).toEqual('Step 3 of 3, unavailable: Step 3');
```
Comment (Blackbaud-SteveBrush) [inline libs/components/tabs/src/lib/modules/tabs/tabset.component.spec.ts:1655]: Good work, testing template logic! 👍

File: libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts [hunk 5]
```diff
@@ -155,4 +179,20 @@ export class SkyTabButtonComponent implements AfterViewInit, OnDestroy {
   public onFocus(): void {
     this.#tabsetService.setFocusedTabBtnIndex(this.tabIndex);
   }
+
+  #updateWizardStepState() {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts]: Needs a return type.

File: libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts (hunk 1 above)
Comment (Blackbaud-ErikaMcVey) [reply libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts:43]: @Blackbaud-SteveBrush I updated this for disabled but this is an internal component we are rendering from the tabset component and the value we pass for active will _always_ be true or false--making it undefined just creates a path that could never happen and therefore is untestable.

**Comment by Blackbaud-ErikaMcVey** [comment]
[AB#2286466](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2286466)

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**